engine/common/requester/engine.go

```diff
@@ -275,11 +275,6 @@ PollLoop:
 
 func (e *Engine) dispatchRequest() (bool, error) {
 
-	e.unit.Lock()
-	defer e.unit.Unlock()
```

Comment:
This locks for too long, we'd like to lock more granular in order to exclude Unicast from the lock period